frontend/service.go

```diff
@@ -313,6 +311,132 @@ func (s *lwdStreamer) GetTaddressBalanceStream(addresses walletrpc.CompactTxStre
 	return nil
 }
 
+// Key is 32-byte txid (as a 64-character string), data is pointer to compact tx.
+var mempoolMap *map[string]*walletrpc.CompactTx
+var mempoolList []string
+
+// Last time we pulled a copy of the mempool from zcashd.
+var lastMempool time.Time
+
+func (s *lwdStreamer) GetMempoolTx(exclude *walletrpc.Exclude, resp walletrpc.CompactTxStreamer_GetMempoolTxServer) error {
```

Comment:
@LarryRuane how does a client know that the transaction it excluded from the fetch because it already fetched it in a previous request is no longer in the mempool?

Reply:
Currently, there isn't a way for lightwalletd to tell you about deletions from the mempool. There are a couple of possible approaches:
- Lightwalletd could return a compact block that can be parsed but that's recognizable in some way as having the special meaning "no longer in mempool." For example, the `spends` and `outputs` arrays are zero-length, or `index` being some impossibly high number. The `txid` would indicate which tx is no longer in the mempool.
- You could expire transactions out of your mempool cache after some time. Of course you won't send its `txid` in the next `GetMempoolTx` request (because you have no knowledge of it), so if the tx is still in the mempool, lwd will return it to you.
- We could add a new gRPC that takes the same `txid` list and returns the subset list of txids that are _not_ in the mempool. You could call that periodically to clear the stale entries out of your mempool cache.
- You should check each incoming compact block, and any transactions it contains that are also in the mempool should be removed from the mempool.